src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -60,6 +60,12 @@ protected Workspace(HostServices host, string workspaceKind)
 
             // queue used for sending events
             var workspaceTaskSchedulerFactory = _services.GetService<IWorkspaceTaskSchedulerFactory>();
```

Comment:
You can just call GetRequiredService instead that will throw for you if it doesn't exist.